tensorflow/python/eager/wrap_function.py

```diff
@@ -277,18 +311,26 @@ def _fetch_preprocesing_callback(f):
 
     fetches = nest.map_structure(_fetch_preprocesing_callback, fetches)
 
-    for f in flat_feeds + tensor_fetches + operation_fetches:
+    for f in flat_feeds + tensor_fetches + operation_fetches \
+        + sparse_tensor_fetches:
       if f.graph is not self._func_graph:
         raise ValueError("Can only prune function whose feeds and fetches "
                          "are from this graph (%s). Input %s is from graph %s" %
                          (self._func_graph, f, f.graph))
     with self._func_graph.as_default():
       pruned_graph = func_graph.FuncGraph(name)
     lift_map = lift_to_graph.lift_to_graph(
-        operation_fetches + tensor_fetches,
+        operation_fetches + tensor_fetches
+        + _sparse_to_dense(sparse_tensor_fetches),
```

Comment:
Slightly more general, can we use expand_composites=True in a nest.flatten and a nest.pack_sequence_as? https://github.com/tensorflow/tensorflow/blob/17266edc52a5a6c8778cf7aafb6bd861769bd60d/tensorflow/python/util/nest.py#L211 and https://github.com/tensorflow/tensorflow/blob/17266edc52a5a6c8778cf7aafb6bd861769bd60d/tensorflow/python/util/nest.py#L396

That should help support new composite types as they get added (and support IndexedSlices and such right now).

Reply:
That is a great idea! I will use `nest.flatten(expand_composites=True)` to get rid of the separate `sparse_tensor_fetches` variable in this function.

It looks like the implementation of `expand_composites` in the `pack_sequence_as` function is currently broken:
```python
In [1]: import tensorflow as tf 
   ...: tf.nest.pack_sequence_as({"foo": 1}, [tf.SparseTensor([[1,1]], [2], [2,3])], expand_composites=True)          

Out[1]: {'foo': <tensorflow.python.framework.sparse_tensor.SparseTensor at 0x112b8f438>}
```

I will work on a separate fix for the problem with `pack_sequence_as`.

Reply:
Isn't the problem there that a SparseTensor (an un-expanded composite) is being passed to pack_sequence_as with expand_composites=True? nest.flatten(..., expand_composites=True) should not return a SparseTensor, so the inverse operation (pack_sequence_as(..., expand_composites=True)) shouldn't have one as input.

Reply:
Although I do see how that could be confusing. I've been interpreting the `expand_composites` argument to `pack_sequence_as` roughly as `pack_composites` (aka "I passed `expand_composites` to nest.flatten so do the opposite of that"). But it's not really clear from the docstring what it's supposed to do.

@edloper is that correct? Should we explain a bit further in the `pack_sequence_as` docstring?

Reply:
Yes, that's right.  `expand_composites=True` basically means "treat composite tensors as if they are structure elements rather than leaves."  I.e., it's roughly like pretending that CompositeTensors are namdetuples whose elements are the tensor components.  So:

```
nest.flatten([t1, SparseTensor(t2, t3, t4)])
```

is roughly equivalent to:

```
nest.flatten([t1, (t2, t3, t4)])
```

and returns `[t1, t2, t3, t4]`.  Going the other way (with `pack_sequence_as`) will pack flat values into the composite tensor by filling in its components.  So:

```
nest.pack_sequence_as([t1, SparseTensor(t2, t3, t4)], [t5, t6, t7, t8])
```

will return `[t5, SparseTensor(t6, t7, t8)]`.

But I'm going to hold off on documenting this for now, since we may actually make a few changes to how CompositeTensors are handled, as part of a change to make it easier to use them everywhere across TensorFlow.  But those changes shouldn't impact this PR -- please go ahead and use nest.flatten and nest.pack_sequence_as with expand_composite=True, and that will make this code work for SparseTensor, RaggedTensor, IndexedSlices, and several other types.

Reply:
Ah, I see now. So I can pass `pack_sequence_as()` a copy of the original `fetches` structure and a flattened list of dense tensors from the copied graph. Then `pack_sequence_as()` will wrap the dense tensors in `SparseTensor` (or other subclasses of `CompositeTensor`) as needed to match the original `fetches` structure. Thanks for the explanation!